src/printer/layers.js

```diff
@@ -107,25 +105,70 @@ function createLayerXYZ(layerSpec, rootFrameState) {
     takeWhile(() => {
       renderer.renderFrame({ ...frameState, time: Date.now() }, context.canvas);
       frameState.tileQueue.reprioritize();
-      frameState.tileQueue.loadMoreTiles(8, 2);
+      frameState.tileQueue.loadMoreTiles(12, 4);
       return frameState.tileQueue.getTilesLoading();
     }, true),
     map(() => {
-      const progress =
-        1 -
-        Object.keys(frameState.tileQueue.queuedElements_).length / tileCount;
-      return progress < 1 ? [progress, null] : [1, context.canvas];
+      let loadedTilesCount = Object.keys(frameState.tileQueue.queuedElements_)
+        .length;
+
+      let progress = 1 - loadedTilesCount / tileCount;
+
+      // this is to make sure all tiles have finished loading before completing layer
+      if (progress === 1 && frameState.tileQueue.getTilesLoading() > 0) {
+        progress -= 0.001;
+      }
+
+      if (progress === 1) {
+        renderer.renderFrame(
```

Comment:
What does it do ? Where was it done before ?

Reply:
Currently the layer is rendered on every update loop (i.e. tile queue update) and this is useless and eats a _lot_ of resources. I realized I had forgotten to remove this call and added a commit for that: https://github.com/camptocamp/inkmap/pull/2/commits/e84c714f95eeb1d129d0b28fa177a3a230ec6868

So now the layer is only rendered once at creation (to enqueue all necessary tiles) and once when all tiles are ready.